cloud_functions/climateiq_data_export_cf/main.py

```diff
@@ -302,20 +352,154 @@ def _calculate_h3_indexes(
+    neighbors = {
+        (x - 1, y + 1),
+        (x, y + 1),
+        (x + 1, y + 1),
+        (x - 1, y),
+        (x + 1, y),
+        (x - 1, y - 1),
+        (x, y - 1),
+        (x + 1, y - 1),
+    }
+    for neighbor_x, neighbor_y in neighbors:
+        if (
+            neighbor_x < 0
+            or neighbor_y < 0
+            or neighbor_x >= study_area_metadata["col_count"]
+            or neighbor_y >= study_area_metadata["row_count"]
+        ):
+            # Chunk is outside the study area boundary.
+            continue
+        query = (
+            chunks_ref.where("x_index", "==", neighbor_x)
+            .where("y_index", "==", neighbor_y)
+            .limit(1)
+        )
+        chunk_doc = query.get()
+        if not chunk_doc.exists:
+            return ValueError(
+                f"Neighbor chunk at index ({neighbor_x, neighbor_y}) \
+                  is missing from the study area."
+            )
+
+        neighbor_chunk_metadata = chunk_doc.to_dict()
+        if (
+            neighbor_chunk_metadata.get("id") is None
+            or neighbor_chunk_metadata.get("row_count") is None
+            or neighbor_chunk_metadata.get("col_count") is None
+            or neighbor_chunk_metadata.get("x_ll_corner") is None
+            or neighbor_chunk_metadata.get("y_ll_corner") is None
+            or neighbor_chunk_metadata.get("x_index") is None
+            or neighbor_chunk_metadata.get("y_index") is None
+        ):
+            raise ValueError(
+                f'Neighbor chunk "{neighbor_chunk_metadata["id"]}" is missing one or \
+                      more required fields: id, row_count, col_count, x_ll_corner, \
+                        y_ll_corner, x_index, y_index'
+            )
+
+        # Determine if the neighbor chunk intersects with any of the boundary H3 cells.
+        neighbor_chunk_boundary = _get_chunk_boundary(
+            study_area_metadata, neighbor_chunk_metadata
+        )
+        intersects = False
+        for h3_boundary in boundary_h3_cells:
+            if h3_boundary.intersects(neighbor_chunk_boundary):
+                intersects = True
+                break
+
+        # Project cell centroids of neighbor chunk to H3 indexes and append any
+        # duplicate H3 indexes to input spatialized_predictions.
+        if intersects:
+            neighbor_chunk_predictions = _read_neighbor_chunk_predictions(
+                bucket_name, object_name, neighbor_chunk_metadata.get("id")
+            )
+            neighbor_chunk_spatialized_predictions = (
+                _build_spatialized_model_predictions(
+                    study_area_metadata,
+                    neighbor_chunk_metadata,
+                    neighbor_chunk_predictions,
+                )
+            )
+            neighbor_chunk_spatialized_predictions[
+                ["h3_index", "h3_centroid_lat", "h3_centroid_lon", "h3_boundary"]
```

Comment:
Nit: I don't think we need to calculate anything but the h3_index for the neighbor chunks. If the operation is expensive, could be worth optimizing here.

Reply:
fair, I added a TODO to consider optimizing this but I'd be surprised if theres are expensive operations. the most expensive operation is probably calculating the H3 index